aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -956,19 +994,58 @@ module aptos_framework::delegation_pool {
     public entry fun set_operator(
         owner: &signer,
         new_operator: address
-    ) acquires DelegationPoolOwnership, DelegationPool, GovernanceRecords {
+    ) acquires DelegationPoolOwnership, DelegationPool, GovernanceRecords, CommissionPercentage {
         let pool_address = get_owned_pool_address(signer::address_of(owner));
         // synchronize delegation and stake pools before any user operation
         // ensure the old operator is paid its uncommitted commission rewards
         synchronize_delegation_pool(pool_address);
         stake::set_operator(&retrieve_stake_pool_owner(borrow_global<DelegationPool>(pool_address)), new_operator);
     }
 
+    /// Allows an owner to update the commission percentage for the operator of the underlying stake pool.
+    public entry fun update_commission_percentage(
+        owner: &signer,
+        new_commission_percentage: u64
+    ) acquires DelegationPoolOwnership, DelegationPool, GovernanceRecords, CommissionPercentage {
+        assert!(new_commission_percentage <= MAX_FEE, error::invalid_argument(EINVALID_COMMISSION_PERCENTAGE));
+        let pool_address = get_owned_pool_address(signer::address_of(owner));
+        assert!(
```

Comment:
Observation: UX changes whether `synchronize_delegation_pool` is called before or after this assert.
Scenario:
1. delegation pool has commission 7% since creation
2. owner sets commission to 7% + 10% at lockup cycle N
3. lockup N ends
4. no operation happened on the delegation pool yet in lockup N+1
6. owner wants to set commission to 7% + 10% + 10% to apply from lockup N+2 which is a valid commission update

- assert then `synchronize_delegation_pool`: tx fails with `ETOO_LARGE_COMMISSION_INCREASE`, but owner can call `synchronize_delegation_pool` separately which makes commission 17%, then try again and now 27% is accepted
- `synchronize_delegation_pool` then assert: commission is firstly updated to 17% and then 27% is accepted directly

Reply:
Now, there is no difference between the two options of different order because `operator_commission_percentage` returns the "effective" commission percentage.